norfair/lib_metrics.py

```diff
@@ -0,0 +1,336 @@
+            )
+
+        indexes = np.argwhere(self.matrix_detections[:, 0] == frame_number)
+        detections = []
+        if len(indexes) > 0:
+            actual_det = self.matrix_detections[indexes]
+            actual_det.shape = [actual_det.shape[0], actual_det.shape[2]]
+            for det in actual_det:
+                points = [[det[2], det[3]], [det[4], det[5]]]
+                points = np.array(points)
+                conf = det[6]
+                new_detection = Detection(
+                    points, np.array([1, 1])
+                )  # set to [1,1] or [conf,conf]
+                detections.append(new_detection)
+        self.actual_detections = detections
+        return detections
+
+
+class Accumulators:
+    def __init__(self):
+        self.matrixes_predictions = []
+        self.paths = []
+
+    def create_acc(self, input_path=None):
+        if not hasattr(self, 'matrixes_predictions'):
+            self.matrixes_predictions=[]
+        if not hasattr(self, 'paths'):
+            self.paths=[]
+
+
+        if input_path is None:
+            raise ValueError(
+                "You must set 'input_path' argument when creating new accumulator"
+            )
+
+        file_name = os.path.split(input_path)[1]
+
+        self.frame_number = 1
+        # save the path of this video in a list
+        self.paths = np.hstack((self.paths, input_path))
+        # initialize a matrix where we will save our predictions for this video (in the MOTChallenge format)
+        self.matrix_predictions = []
+
+        # initialize progress bar
+        seqinfo_path = os.path.join(input_path, "seqinfo.ini")
+        length = search_value_on_document(seqinfo_path, "seqLength")
+        self.progress_bar_iter = track(
+            range(length - 1), description=file_name, transient=False
+        )
+
+    def update(self, predictions=None):
+        # get the tracked boxes from this frame in an array
+        for obj in predictions:
+            new_row = [
+                self.frame_number,
+                obj.id,
+                obj.estimate[0, 0],
+                obj.estimate[0, 1],
+                obj.estimate[1, 0] - obj.estimate[0, 0],
+                obj.estimate[1, 1] - obj.estimate[0, 1],
+                -1,
+                -1,
+                -1,
+                -1,
+            ]
+            if np.shape(self.matrix_predictions)[0] == 0:
+                self.matrix_predictions = new_row
+            else:
+                self.matrix_predictions = np.vstack((self.matrix_predictions, new_row))
+        self.frame_number += 1
+        # advance in progress bar
+        try:
+            next(self.progress_bar_iter)
+        except StopIteration:
+            self.matrixes_predictions.append(self.matrix_predictions)
+            return
+
+    def compute_metrics(
+        self,
+        metrics=None,
+        generate_overall=True,
+    ):
+        if metrics == None:
```

Comment:
As a general rule we should compare to `None` using `is` instead of `==`. Justification if you're interested: http://jaredgrubb.blogspot.com/2009/04/python-is-none-vs-none.html

Reply:
Ok